ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl/Pure.hs

```diff
@@ -0,0 +1,321 @@
+      | let firstTxSize = txSize firstTx
+            curSize = msNumBytes $ isMempoolSize is
+      , curSize + firstTxSize > getMempoolCapacityBytes (isCapacity is)
+      = Done toAdd
+
+      | otherwise
+      = maybeWriteState getValidationOutcome
+        where
+            (eVtx, vr) = extendVRNew cfg firstTx txSize $ validationResultFromIS is
+
+            is' = internalStateFromVR vr
+
+            maybeWriteState = if null (vrNewValid vr)
+                              then id
+                              else
+                                -- Each time we have found a valid transaction,
+                                -- we update the Mempool. This keeps our STM
+                                -- transactions short, avoiding repeated work.
+                                --
+                                -- Note that even if the transaction were
+                                -- invalid, we could still write the state,
+                                -- because in that case we would have that @is
+                                -- == is'@, but there's no reason to do that
+                                -- additional write.
+                                WriteTx is'
+
+            -- We only extended the ValidationResult with a single transaction
+            -- ('firstTx'). So if it's not in 'vrInvalid', it must be in
+            -- 'vrNewValid'.
+            getValidationOutcome = case eVtx of
+              Right vtx ->
+                assert (isJust (vrNewValid vr)) $
+                Trace (TraceMempoolAddedTx
+                        vtx
+                        (isMempoolSize is)
+                        (isMempoolSize is')) $
+                Processed (MempoolTxAdded vtx) $ go toAdd' is'
+              Left err ->
+                assert (isNothing (vrNewValid vr))  $
+                assert (length (vrInvalid vr) == 1) $
+                Trace (TraceMempoolRejectedTx
+                      firstTx
+                      err
+                      (isMempoolSize is)) $
+                Processed (MempoolTxRejected firstTx err) $
+                go toAdd' is'
+
+-- | A datatype containing the state resulting after removing the requested
+-- transactions from the mempool and maybe a message to be traced while removing
+-- them.
+data RemoveTxs blk = WriteRemoveTxs (InternalState blk) (Maybe (TraceEventMempool blk))
+
+-- | Intepret a 'RemoveTxs' with the resulting values produced by manually
+-- removing the transactions given to 'pureRemovTxs' from the mempool.
+runRemoveTxs :: forall m blk. IOLike m
+  => StrictTVar m (InternalState blk)
+  -> RemoveTxs blk
+  -> STM m (Maybe (TraceEventMempool blk))
+runRemoveTxs stateVar (WriteRemoveTxs is t) = do
+  writeTVar stateVar is
+  return t
+
+-- | Craft a 'RemoveTxs' that manually removes the given transactions from the
+-- mempool, returning inside it an updated InternalState.
+pureRemoveTxs
+  :: ( LedgerSupportsMempool blk
```

Comment:
Style: the signatures in this module are inconsistent regarding `::` placement and indentation. Please make them uniform. (See `StyleGuide.md` for the preferred details, though "consistency within the module" is the only hard requirement.)

Reply:
I look forward to some kind of automation on this stuff, as I have to consult the guide each time as it is still not wired in my brain.

Fixed (resolving)